objectstore/objectstore.go

```diff
@@ -45,9 +48,20 @@ func Open(name string) (*Store, error) {
 	return &Store{objectstore: o}, nil
 }
 
-// Lookup fetches a key from the associated object store.
+// Lookup fetches a key from the associated object store.  If the key does not
+// exist, Lookup returns the sentinel error ErrKeyNotFound.
 func (s *Store) Lookup(key string) (*Entry, error) {
 	val, err := s.objectstore.Lookup(key)
+	if err != nil {
+
+		// turn FastlyStatusNone into NotFound
+		if status, ok := fastly.IsFastlyError(err); ok && status == fastly.FastlyStatusNone {
+			return nil, ErrKeyNotFound
```

Comment:
I realize now that returning `nil` for the `Entry` changes behaviour for keys that don't exist.  Previously we would simply return an empty body (which was fine to read from), but. now we're returning a nil pointer than will segfault if you don't check for it.  I guess what's more likely to happen is that people will error out in places they wouldn't before, since most people are likely checking the error return, rather than having people *not* checking the error and blindly proceeding and now we're going to `panic` with a `nil` interface read.